addons/l10n_ro_edi_stock/models/l10n_ro_edi_stock_etransport_interface.py

```diff
@@ -0,0 +1,640 @@
+        })
+
+    def _create_document_sending_failed(self, message: str, load_id: str | None = None, uit: str | None = None):
+        self.ensure_one()
+        return self.env['l10n_ro.edi.stock.document'].create({
+            'interface_id': self.id,
+            'state': 'sending_failed',
+            'message': message,
+            'load_id': load_id,
+            'uit': uit,
+        })
+
+    def _create_document_sent(self, load_id: str, uit: str):
+        self.ensure_one()
+        return self.env['l10n_ro.edi.stock.document'].create({
+            'interface_id': self.id,
+            'state': 'sent',
+            'load_id': load_id,
+            'uit': uit,
+        })
+
+    def _get_failed_documents(self):
+        self.ensure_one()
+        return self.l10n_ro_edi_stock_document_ids.filtered(lambda doc: doc.state == 'sending_failed')
+
+    def _get_sending_and_failed_documents(self):
+        self.ensure_one()
+        return self.l10n_ro_edi_stock_document_ids.filtered(lambda doc: doc.state in ('sending_failed', 'sending'))
+
+    def _get_last_sent_document(self):
+        self.ensure_one()
+        sent_documents = self.l10n_ro_edi_stock_document_ids.filtered(lambda doc: doc.state == 'sent').sorted()
+
+        if sent_documents:
+            return sent_documents[0]
+
+    def _get_current_document(self):
+        self.ensure_one()
+        return self.l10n_ro_edi_stock_document_ids.sorted()[0] if self.l10n_ro_edi_stock_document_ids else None
+
+    # Send Logic
+
+    def _send_etransport_document(self, send_type: str):
+        """
+        Send the eTransport document to anaf
+        :param send_type: 'send' (initial sending of document) | 'amend' (correct the already sent document)
+        """
+        self.ensure_one()
+
+        if errors := self._validate_data():
+            self._get_failed_documents().unlink()
+
+            if send_type == 'amend':
+                last_sent_document = self._get_last_sent_document()
+                self._create_document_sending_failed(message='\n'.join(errors), load_id=last_sent_document.load_id, uit=last_sent_document.uit)
+            else:
+                self._create_document_sending_failed(message='\n'.join(errors))
+            return
+
+        template = self.env['ir.ui.view']._render_template('l10n_ro_edi_stock.l10n_ro_template_etransport', values={
+            'send_type': send_type,
+            'interface': self,
+            'STATE_CODES': STATE_CODES,
+            'declarant_ref': self._get_declarant_ref(),
+            'partner': self._get_commercial_partner(),
+            'transport_partner': self._get_transport_partner(),
+            'scheduled_date': self._get_scheduled_date(),
+            'vehicle_number': self.l10n_ro_edi_stock_vehicle_number.upper(),
+            'trailer1': (self.l10n_ro_edi_stock_trailer_1_number or '').upper() or None,
+            'trailer2': (self.l10n_ro_edi_stock_trailer_2_number or '').upper() or None,
+        })
```

Comment:
as I said in the earlier comments, let's compute as much values here instead of in the XML :)

note to self: review stopped here